pkg/reconciler/service/service.go

```diff
@@ -110,6 +120,15 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, service *v1.Service) pkg
 	}
 
 	c.checkRoutesNotReady(config, logger, route, service)
+
+	extReady, err := c.extension.UpdateExtensionStatus(ctx, service)
+	if err != nil {
+		return err
+	}
+
+	if !extReady {
+		c.enqueueAfter(service, time.Duration(60*float64(time.Second)))
+	}
```

Comment:
What's the reason to re-enqueue the service?

Reply:
This c.enqueueAfter schedules another call of the reconciliation loop after a certain time as specified as the parameter. Since `UpdateExtensionStatus` returns the status of the extension, if the status `extReady` is false, meaning the extension is not ready, we need to schedule another call of the reconciliation loop to check the status again and again until it reaches ready.

Reply:
Do you have an implementation of the extension for the orchestrator?